### DISCUSSION TITLE
Plädoyer für volle Dezentralität

### DISCUSSION THREAD

**Discussion by KampfCaspar** [e-id-admin/governance-sounding-board#23] [General]
Am heutigen Webex wurden verschiedene 'Zentralisierungen' erwähnt oder implizit vorausgesetzt. So z.B.:
1. Es gebe eine zentrale Registry (im Sinne des Technical Trust, vgl. #15 )
2. Es müsse zentral entschieden werden, wie weit das Ökosystem gehe - konkret ob man tatsächlich die Mitgliederausweise aller Vereine aufnehmen wolle.

Um ein möglichst lebendiges Ökosystem zu ermöglichen, möchte ich deshalb ein Plädoyer für _volle_ Dezentralität platzieren. Das 'System Schweizer eID' soll sich IMHO voll auf die Definition von Standards und Referenzimplementierungen stützen - völlig ohne zwingendes, zentrales Element.

Auf diese Standards aufbauend kann dann vom Bund die 'Kern-ID' (also das Pendant zur physischen ID) implementiert werden. Dafür braucht es ein konkretes Trust-Directory und eine konkrete Registry. Das sind dann aber nur 'eine unter vielen'!

Völlig unabhängig von der Bundes-Infrastruktur kann der 'Bibliotheksverband Bern', SWITCH mit eduID, Vereine bzw. Verbände oder auch Familie Müller familienintern das Framework verwenden. Diese sind nicht gezwungen, ein zentrales Trust Directory oder auch nur eine bestimmte Registry zu nutzen. Familie Müller braucht vielleicht überhaupt keine Registry, sondern installiert Trust Anchors sowie Revokations-Daten manuell direkt auf ihren eigenen Geräten. Es braucht auch keinerlei politische Entscheide, wer das Framework wie verwenden darf.

Sauber definiert besteht auch kein Problem mit Abhängigkeiten zwischen verschiedenen Trust Directories und Registries. Es ist problemlos möglich, dass der Mitarbeiterausweis (bzw. sein Trust Directory und seine Registry) für den Personennamen auf das Credential der Kern-ID (mit ihrem eigenen Trust Directory und Registry) verweist.

In dieser Konzeption bleibt es selbstverständlich möglich, dass der Bund auch eine für alle (optional) nutzbare Registry mit Public-Keys und Revokationsdaten betreibt - die müsste dann aber völlig offen sein, weil ja eben kein 'Human Trust' dahinter steht. Je nach Issuer kann so ein vielgenutzter 'Distributed Hash Table' sogar datenschutzrechtlich vorteilhaft sein, weil einzelne Zugriffe weniger rekonstruierbar sind. Es gibt aber keinen Grund, dessen Verwendung vorzuschreiben.

In diesem Sinne: Define the interoperability (standards) and set it free! ;)

**Comment by grt-dti** [comment]
@KampfCaspar: Den ersten Reflex unterdrückend will ich verstehen was präzise du meinst. "volle Dezentralität" kann ja vieles sein:
- volle Dezentralität der Holder: jeder Holder kann (Fähigkeit) so viele Wallets haben wie sein Dezentralitätsbedarf bestimmt (Pseudonymität der Holder) - und alles passt noch zusammen; alles kann damit umgehen
- volle Dezentralität der Verifier: jeder Verifier kann in so vielen "Identitäten" auftreten wie er will und VC's validieren (Pseudonymität der Verifier) - und alles passt noch zusammen; alles kann damit umgehen
- volle Dezentralität der Issuer: jeder Issuer kann in so vielen "Identitäten" auftreten und VC's ausgeben wie er/sie/es nach seinen/ihrem Dezentralitätsbedarf notwendig erscheint (Pseudonymität der Issuer) - und alles passt noch zusammen; alles kann damit umgehen
- volle Dezentralität der Registries: es gibt eine Vielzahl von Registries und jeder Registry (Issuer) kann (mehr oder weniger frei) entscheiden, bei welcher er/sie/es sich registriert - und alles passt noch zusammen; alles kann damit umgehen

Welche (gern auch mehrere) davon meintest du und WARUM sind diese "vollen Dezentralität"en wichtiger als die die du nicht meinst? WARUM plädierst du nicht für die Anderen?
Nur damit wir uns richtig verstehen: dies ist eine Frage; keine Meinung für oder gegen dein Plädoyer - ich will's erstmal verstehen bevor ich mir meine Meinung bilde.....

**Reply by KampfCaspar** [reply]
Nur weil ich (hier) für eine Art der Dezentralität plädiere, heisst das nicht, dass ich andere ablehne. Hier geht es mir darum, dass möglichst keinerlei Infrastruktur (physisch oder logisch) für das Framework zwingend sein sollte. Die 'Kern-ID' (der Ersatz für die physische Karte) braucht natürlich etwas zentrales - aber wenn man das konsequent als 'eine Anwendung unter vielen' konzipiert, bleibt möglichst viel Freiheit.

**Reply by grt-dti** [reply]
> Nur weil ich (hier) für eine Art der Dezentralität plädiere, heisst das nicht, dass ich andere ablehne. Hier geht es mir darum, dass möglichst keinerlei Infrastruktur (physisch oder logisch) für das Framework zwingend sein sollte. Die 'Kern-ID' (der Ersatz für die physische Karte) braucht natürlich etwas zentrales - aber wenn man das konsequent als 'eine Anwendung unter vielen' konzipiert, bleibt möglichst viel Freiheit.

@KampfCaspar das ist eine "schöne" Antwort, aber noch keine Antwort auf meine Frage - ich will destillieren, WO es WELCHE Dezentralität "braucht" / haben soll und WARUM diese "besser" ist als andere.....

**Comment by christianheimann** [comment]
Danke für den Input @KampfCaspar; hilft uns zu sehen, wo wir nicht klar genug waren. Zur Komplettierung des Bildes:

<img src="https://user-images.githubusercontent.com/12694135/156721778-2db30002-0de5-4ccd-b741-49e8b6c33bf6.png" width="500">

Die angesprochene Registry, wird in heute umgesetzten SSI-Umgebungen nach meiner Kenntnis stets als gemeinsam genutzte Basis-Infrastruktur gebaut; auch weil eine zusätzliche Fragmentierung keinen Zusatznutzen bringt, da es sich um ein öffentlich lesbares Element handelt, das u.a. auch für die Entkopplung von Issuer und Verifier dient. Eine gemeinsam genutzte Registry aber es erlaubt allen möglichen Akteuren, darauf aufzubauen - und sich nicht mit der "sowieso nötigen und für alle gleichen Basis" auseinandersetzen zu müssen.

"Zentral" wurde im Speech also eher verwendet im Sinne von "wichtig" und "gemeinsam nutzbar", unabhänig von dessen technischen Ausführung ob auf einem einzelnen Server betrieben oder in einem verteilten Netzwerk.

### Ergänzung zu:

> Es müsse zentral entschieden werden, wie weit das Ökosystem gehe - konkret ob man tatsächlich die Mitgliederausweise aller Vereine aufnehmen wolle.

Das wurde nicht ganz eindeutig dargestellt. Bei der Registry gehen wir vom Projektteam heute von einem offenen System aus, es wird aber trotzdem eine Governance brauchen. Diese Governance gilt es gemeinsam zu entwickeln, u.a. im Gesetzgebungsprozess und hier in dieser Community.

> Auf diese Standards aufbauend kann dann vom Bund die 'Kern-ID' (also das Pendant zur physischen ID) implementiert werden. Dafür braucht es ein konkretes Trust-Directory und eine konkrete Registry. Das sind dann aber nur 'eine unter vielen'!

Das würde ich bestätigen, das kann eine unter vielen sein. Es wird nicht ausgeschlossen, dass es weitere Registries geben könnte; Interoperabilität ist sicher ein Thema. Es sollte und wird wohl keine Einschränkungen geben, so etwas machen _zu dürfen_. Genauso wird jeder frei sein, auf den gleichen Standards "sein Ding" aufbauen zu dürfen.

Einer der Kerngedanken der Vertrauensinfrastruktur von digitalen Nachweisen ist aber genau das **organisationsübergreifende** Element, wo es nicht um "sein Ding" geht, sondern um die Möglichkeit vertrauensvolle Nachweise über jenste Organisations- und Systemgrenzen hinweg zu ermöglichen.

> Familie Müller braucht vielleicht überhaupt keine Registry, sondern installiert Trust Anchors sowie Revokations-Daten manuell direkt auf ihren eigenen Geräten.

Im Verständnis, dass Registry und Trust Directory unterschiedliche Aufgaben haben, braucht es zwingend die Informationen einer Registry und optional Informationen eines Trust Directory. Der Mechanismus, manuell auf eigenen Geräten Vertrauens-Beweise zu verwalten und "aktuell" zu bleiben ist das, was in meinen Augen die breite Nutzung behindert ‒ sonst würden wir heute auch alle verschlüsselte E-Mails versenden... Es braucht Convenience zur Massenadoption. Technologische Umsetzungen von SSI versprechen diese Convenience.

### Plädoyer für eine Gemeinsame Basis-Infrastruktur
Die Gelegenheit, welche sich aktuell bietet, ist eine riesige Chance für die Schweiz, eine gemeinsame Basis-Infrastruktur (Wallet, Registry, ggf. Agent) zu schaffen -- nicht nur Standards festzuschreiben. Diese Infrastruktur könnten dann alle nutzen, _die das möchten_, in den verschiedenen Rollen Issuer, Holder und Verifier. Basierend auf offenen und möglichst gut nutzbaren Standards würde es aber auch möglich sein, anderweitige Ökosysteme über die gleichen Mechanismen einzubinden oder zu nutzen (bspw. EU aber auch die von @KampfCaspar angesprochenen von der Bundesinfrastruktur unabhängigen Ökosysteme). Das eine schliesst das andere nicht aus.

**Reply by KampfCaspar** [reply]
> Einer der Kerngedanken der Vertrauensinfrastruktur von digitalen Nachweisen ist aber genau das **organisationsübergreifende** Element,

Sehr einverstanden - aber das bedingt eben _nicht_ eine (zwingend) einheitliche Registry. Sauber definiert können auch organisations- bzw. registryübergreifend Abhängigkeiten abgebildet werden: Der Mitarbeiterausweis kann auf einer eigenen Registry basieren und trotzdem konkret für das Credential zum Namen auf die Registry des Bundes zurückgreifen.

Die eine, grosse Registry braucht es nur bei 'anarchischen' Self-Sovereign-Identity-System ohne jede Authorität, ohne jeden Staat. Diese (fiktive) Idee ist aber nicht das, was die Schweiz aufbauen möchte. Im diskutierten Modell steht _immer_ irgendein Human Trust dahinter.

**Reply by grt-dti** [reply]
> > Einer der Kerngedanken der Vertrauensinfrastruktur von digitalen Nachweisen ist aber genau das **organisationsübergreifende** Element,
> 
> Sehr einverstanden - aber das bedingt eben _nicht_ eine (zwingend) einheitliche Registry. Sauber definiert können auch organisations- bzw. registryübergreifend Abhängigkeiten abgebildet werden: Der Mitarbeiterausweis kann auf einer eigenen Registry basieren und trotzdem konkret für das Credential zum Namen auf die Registry des Bundes zurückgreifen.
> 
> Die eine, grosse Registry braucht es nur bei 'anarchischen' Self-Sovereign-Identity-System ohne jede Authorität, ohne jeden Staat. Diese (fiktive) Idee ist aber nicht das, was die Schweiz aufbauen möchte. Im diskutierten Modell steht _immer_ irgendein Human Trust dahinter.

sehr schöne Idee und valabler Ansatz. Wie bringen wir hier die Interoperabilität zwischen den Registries (die ja ihren DLT selbstständig wählen) hin? Diese "grosse Freiheit" ist halt immer eine grosse Herausforderung für die Interoperabilität- (nicht nur die technisch/technologische, sondern auch für geschäftliche, ökonomische und juristische Aspekte)

**Comment by pascalgottret** [comment]
Ich habe mir dazu mal ein Paar Gedanken gemacht, auch weil es beim letzten Meeting am 7.4. auf dem Menti Board viele Diskussionen rund um das Thema gab.

Die Frage nach der VDR (Verifiable Data Registry) ist für mich eine rein **technische:**

Grundsätzlich muss die VDR für jeden Teilnehmer des Ökosystems frei auswählbar sein. Selbst wenn für das Ambitionsniveau 3 eine spezifische VDR (z.B. Hyperledger) vorgeschrieben wird, ist es sicher, dass andere Ökosysteme im Ausland auf anderen VDR aufbauen. Das ist völlig legitim.

Daher ist die Interoperabilität extrem wichtig, damit wir uns nicht isolieren. Die Wallets, mit dem Verifier arbeiten, müssen sicherstellen, dass alle möglichen DID-Resolver unterstützt werden. Die DIF hat bereits eine Lösung dafür bereitgestellt, die eine enorme Breite von DID Methods unterstützt (https://github.com/decentralized-identity/universal-resolver). Es können jederzeit weitere beantragt werden. 

Ich bin daher der Meinung, dass unser Ökosystem Empfehlungen zu VDR macht und Tools für eine möglichst einfache Adaptierung bereitstellt. Aber von einer Vorschrift eines spezifischen VDR würde ich absehen. Die Interoperabilität müssen die Wallets sicherstellen.

Auf welcher Basis die VDR basiert ist zweitrangig, Hauptsache ist, dass ein DID Resolving ausserhalb des Ökosystems unterstützt wird. Ich bin allerdings trotzdem der Meinung, dass ein VDR öffentlich, Transaktionen gebührenfrei und skalierbar sein muss, damit ein möglichst offenes System entstehen kann.  Davon gibt heute relativ wenige (und Hyperledger ist m.E. nicht optimal; selbst die EU schaut mit dem EBSI Projekt langfristig nach anderen Lösungen).

Die Frage nach den **geschäftlichen, ökonomischen und juristischen** Aspekten ist völlig richtig, aber nicht mit VDR sondern mit dem Governance Framework zu beantworten (über die Trennung Trust registry und VDR hat bereits Vitus in einem anderen Thread darauf aufmerksam gemacht). So wie ich das verstanden habe, wird eine Arbeitsgruppe von Digitalswitzerland demnächst ein Paper ausgeben, wie ein solche Framework aussehen könnte. Bin gespannt auf den Inhalt.
